include/bluetooth/services/ots.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef BT_GATT_OTS_H_
+#define BT_GATT_OTS_H_
+
+/**@file
+ * @defgroup bt_gatt_ots Object Transfer Service API
+ * @{
+ * @brief API for the Object Transfer Service (OTS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+#include <bluetooth/conn.h>
+
+/**@brief Type of an OTS object. */
+struct bt_gatt_ots_obj_type {
+	/* Bluetooth UUID type */
+	u8_t uuid;
+
+	union {
+		/* 16-bit UUID value */
+		u16_t uuid_16;
+
+		/* 128-bit UUID value */
+		u8_t  uuid_128[16];
+	};
+};
+
+/**@brief Properties of an OTS object. */
+struct bt_gatt_ots_obj_props {
+	union {
+		/* Properties as bitfields */
+		struct {
+			/** Bit 0 Deletion of this object is permitted */
+			bool delete_permitted	: 1;
+			/** Bit 1 Execution of this object is permitted */
+			bool execute_permitted	: 1;
+			/** Bit 2 Reading this object is permitted */
+			bool read_permitted	: 1;
+			/** Bit 3 Writing data to this object is permitted */
+			bool write_permitted	: 1;
+			/** Bit 4 Appending data to this object that increases
+			 *	  its Allocated Size is permitted
+			 */
+			bool append_permitted	: 1;
+			/** Bit 5 Truncation of this object is permitted */
+			bool truncate_permitted	: 1;
+			/** Bit 6 Patching this object by overwriting some
+			 *	  of the object's existing contents is permitted
+			 */
+			bool patch_permitted	: 1;
+			/** Bit 7 This object is a marked object */
+			bool is_marked		: 1;
+		} decoded;
+
+		/* Properties as raw value */
+		u32_t raw;
+	};
+};
+
+/**@brief Descriptor for OTS object intialization. */
+struct bt_gatt_ots_obj_init {
+	/* Object Name */
+	char                        *name;
+
+	/* Object Type */
+	struct bt_gatt_ots_obj_type  type;
+
+	/* Object Size (allocated and current) */
+	u32_t                        size;
+
+	/* Object Properties */
+	struct bt_gatt_ots_obj_props props;
+};
+
+/** @brief OTS server callback structure. */
+struct bt_gatt_ots_cb {
```

Comment:
I was wondering if would be necessary to have a CB that lets the application know if an object is currently being read. Right now there's only obj_read which is called when the read has finished. 

The reasoning of why it might make sense to have "read started", is so that the application above can ensure that the data isn't changed in mid-read. This is especially important if the data length becomes smaller (although that isn't "possible" in this API)